cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -230,13 +228,19 @@ else if ( systemBookmark != null )
     }
 
     private ThrowingAction<CommandException> getPing() {
-        String query =
-                activeDatabaseNameAsSetByUser.compareToIgnoreCase(SYSTEM_DB_NAME) == 0 ? "CALL db.indexes()" : "RETURN 1";
+
         return () -> {
-            Result run = session.run(query.trim());
             ResultSummary summary = null;
+            Result run = session.run( "CALL db.ping()");
             try {
                 summary = run.consume();
+            } catch (ClientException e) {
+                //In older versions there is no db.ping procedure, use legacy method.
+                if ( procedureNotFound( e ) ) {
+                    run = session.run(isSystemDb() ? "CALL db.indexes()" : "RETURN 1" );
+                } else {
```

Comment:
Are you sure we really need this second round-trip and that we do not get what we need in the result summary anyway? Could it be that we even get the relevant exception for database offline etc. just by trying and catching this? An integration test would be useful to verify this.

Reply:
I'll have a look. It is a tricky integration test right since it will depend on the server version? Though all of our integration tests touch this path so I guess if we remove the extra roundtrip and everything works on all versions we kind of test it?

Reply:
Hmm yes it is unfortunate that the driver API forces us to get a completed `ResultSummary` to get the information we need (at least version seem to be available internally in the driver after the protocol negotiation and the default database name is not super important to get immediately on connection). 

I would prefer us to do `Result run = session.run( isSystemDb() ? "CALL db.ping()" : "RETURN 1" )` in the first instance, and then we can retry with `run = session.run( "CALL db.indexes()" )` on procedure exception. This way it wouldn't affect operation toward a normal database.